services/src/api/model/db_types.rs

```diff
@@ -607,6 +616,960 @@ impl<S: std::hash::BuildHasher + std::default::Default> From<HashMapTextTextDbTy
+            } => Ok(TypedGeometry::MultiPoint(multi_point)),
+            TypedGeometryDbType {
+                data: false,
+                multi_point: None,
+                multi_line_string: Some(multi_line_string),
+                multi_polygon: None,
+            } => Ok(TypedGeometry::MultiLineString(multi_line_string)),
+            TypedGeometryDbType {
+                data: false,
+                multi_point: None,
+                multi_line_string: None,
+                multi_polygon: Some(multi_polygon),
+            } => Ok(TypedGeometry::MultiPolygon(multi_polygon)),
+            _ => Err(Error::UnexpectedInvalidDbTypeConversion),
+        }
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "OgrSourceDataset")]
+pub struct OgrSourceDatasetDbType {
+    pub file_name: String,
+    pub layer_name: String,
+    pub data_type: Option<VectorDataType>,
+    pub time: OgrSourceDatasetTimeType,
+    pub default_geometry: Option<TypedGeometry>,
+    pub columns: Option<OgrSourceColumnSpec>,
+    pub force_ogr_time_filter: bool,
+    pub force_ogr_spatial_filter: bool,
+    pub on_error: OgrSourceErrorSpec,
+    pub sql_query: Option<String>,
+    pub attribute_query: Option<String>,
+    pub cache_ttl: CacheTtlSeconds,
+}
+
+impl From<&OgrSourceDataset> for OgrSourceDatasetDbType {
+    fn from(other: &OgrSourceDataset) -> Self {
+        Self {
+            file_name: other.file_name.to_string_lossy().to_string(),
+            layer_name: other.layer_name.clone(),
+            data_type: other.data_type,
+            time: other.time.clone(),
+            default_geometry: other.default_geometry.clone(),
+            columns: other.columns.clone(),
+            force_ogr_time_filter: other.force_ogr_time_filter,
+            force_ogr_spatial_filter: other.force_ogr_spatial_filter,
+            on_error: other.on_error,
+            sql_query: other.sql_query.clone(),
+            attribute_query: other.attribute_query.clone(),
+            cache_ttl: other.cache_ttl,
+        }
+    }
+}
+
+impl TryFrom<OgrSourceDatasetDbType> for OgrSourceDataset {
+    type Error = Error;
+
+    fn try_from(other: OgrSourceDatasetDbType) -> Result<Self, Self::Error> {
+        Ok(Self {
+            file_name: other.file_name.into(),
+            layer_name: other.layer_name,
+            data_type: other.data_type,
+            time: other.time,
+            default_geometry: other.default_geometry,
+            columns: other.columns,
+            force_ogr_time_filter: other.force_ogr_time_filter,
+            force_ogr_spatial_filter: other.force_ogr_spatial_filter,
+            on_error: other.on_error,
+            sql_query: other.sql_query,
+            attribute_query: other.attribute_query,
+            cache_ttl: other.cache_ttl,
+        })
+    }
+}
+
+#[derive(Debug, ToSql, FromSql)]
+#[postgres(name = "OgrMetaData")]
+pub struct OgrMetaDataDbType {
+    pub loading_info: OgrSourceDataset,
```

Comment:
why not use the `OgrSourceDatasetDbType` here?

Reply:
I want to use them as few as possible. As long as a type implements FromSql/ToSql, we should use it.

I guess we discussed this already before :smile: .

In the worst case, I would have to create extra DbTypes to use them as fields. By doing it this way, we keep it as minimal as possible.